File: go/transmit/response.go
package transmit

import "encoding/json"

type apiEnvelope struct {
	Success bool            `json:"success"`
	Message string          `json:"message"`
	Data    json.RawMessage `json:"data"`
}

func unwrapData(raw json.RawMessage) json.RawMessage {
	var env apiEnvelope
	if err := json.Unmarshal(raw, &env); err != nil {
		return raw
	}
	if env.Data != nil {
		return env.Data
	}
	return raw
}

func decodeData[T any](raw json.RawMessage, out *T) error {
	return json.Unmarshal(unwrapData(raw), out)
}

func decodeList[T any](raw json.RawMessage) ([]T, error) {
	data := unwrapData(raw)

	var tuple []json.RawMessage
	if err := json.Unmarshal(data, &tuple); err == nil && len(tuple) > 0 {
		var probe []T
		if err := json.Unmarshal(tuple[0], &probe); err == nil {
			return probe, nil
		}
	}

	var items []T
	if err := json.Unmarshal(data, &items); err != nil {
		return nil, err
	}
	return items, nil
}

File: python/transmit/response.py
from typing import Any, List, Optional, Tuple, TypeVar

T = TypeVar("T")


def unwrap_data(raw: Any) -> Any:
    if isinstance(raw, dict) and "success" in raw and "data" in raw:
        return raw.get("data")
    return raw


def unwrap_list(raw: Any) -> List[Any]:
    data = unwrap_data(raw)
    if isinstance(data, list) and data and isinstance(data[0], list):
        return data[0]
    if isinstance(data, list):
        return data
    return []


def unwrap_paginated(raw: Any) -> Tuple[List[Any], Optional[int]]:
    data = unwrap_data(raw)
    if isinstance(data, list) and data and isinstance(data[0], list):
        total = data[1] if len(data) > 1 and isinstance(data[1], int) else None
        return data[0], total
    return unwrap_list(raw), None
